OpenXLSX/CMakeLists.txt

```diff
@@ -185,6 +185,7 @@ if ("${OPENXLSX_LIBRARY_TYPE}" STREQUAL "SHARED")
 
 endif()
 
+target_compile_definitions(OpenXLSX PUBLIC OPENXLSX_CMAKE)
```

Comment:
To address (4), I added a flag to indicate that the build was performed using CMake.